### PR TITLE
Identity Contract: `create_identity`, `add_address` and boilerplate for other required functions 

### PR DESCRIPTION
**Pull request by Szegoo**
This PR:
- adds all the necessary storage values that are needed in the identity contract.
- implements the `create_identity` function and covers it with tests
- implements the `add_address` function and covers it with tests
- adds unimplemented functions for identity management. These will be implemented in separate PRs

Part of: #2

### CODE REVIEW

File: contracts/identity/lib.rs [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
 
+use ink::storage::traits::StorageLayout;
+
+/// Each identity will be associated with a unique identifier called `IdentityNo`.
+pub type IdentityNo = u64;
+
+/// We want to keep the address type very generic since we want to support any
+/// address format. We won't actually keep the addresses in the contract itself.
+/// Before storing them, we'll encrypt them to ensure privacy.
+pub type Address = Vec<u8>;
```
Comment (cuteolaf) [inline contracts/identity/lib.rs:23]: I think we should limit the length of an Address

File: contracts/identity/lib.rs [hunk 2]
```diff
@@ -1,14 +1,51 @@
...
+	Kusama(String),
+	/// Rococo network
+	Rococo(String),
+}
+
+#[derive(scale::Encode, scale::Decode, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo, StorageLayout))]
+pub struct IdentityInfo {
+	/// Each address is associated with a specific blockchain.
+	addresses: Vec<(Network, Address)>,
```
Comment (cuteolaf) [inline contracts/identity/lib.rs:32]: I would prefer use of Mapping for time efficiency

Comment (Szegoo) [reply contracts/identity/lib.rs:32]: Yes, that is what I was thinking also, but it is not possible: https://substrate.stackexchange.com/questions/1659/how-to-have-a-mapping-in-a-custom-structure-inside-an-ink-contract

Comment (cuteolaf) [reply contracts/identity/lib.rs:32]: Is it possible to declare a storage item like `mapping(IdentityNo, mapping(Network, Address))`?

Comment (Szegoo) [reply contracts/identity/lib.rs:32]: No, I have tried.

Comment (cuteolaf) [reply contracts/identity/lib.rs:32]: okay, got it.
seems like we have no other choice than using Vec.

Comment (Szegoo) [reply contracts/identity/lib.rs:32]: Yes, that just seems like the most reasonable choice we have.

File: contracts/identity/lib.rs (hunk 1 above)
Comment (Szegoo) [reply contracts/identity/lib.rs:23]: #8

**Review by cuteolaf — APPROVED** [review]

**Pull request merged**